api/http/handler/settings/settings_update.go

```diff
@@ -95,6 +96,9 @@ func (payload *settingsUpdatePayload) Validate(r *http.Request) error {
 		}
 	}
 
+	if payload.OAuthSettings.AuthStyle < oauth2.AuthStyleAutoDetect || payload.OAuthSettings.AuthStyle > oauth2.AuthStyleInHeader {
```

Comment:
what was the default value? I think choosing a default is better than requiring it, so we keep the API backward compatiable

Reply:
ok, if the parameter is not supplied it will be 0, hence the default. 
error will be returned if it's invalid (e.g -1)